server/graphql/loaders.js

```diff
@@ -55,7 +55,15 @@ export const loaders = req => {
   return {
     expense: {
       findById: new DataLoader(ids =>
-        models.Expense.findAll({ where: { id: { [Op.in]: ids } } }).then(results => sortResults(ids, results, 'id')),
+        models.Expense.findAll({ where: { id: { [Op.in]: ids } } }).then(results => {
+          const viewer = req.remoteUser;
+          return sortResults(ids, results, 'id').map(result => {
+            if (!result) return null;
+            if (viewer && viewer.isAdmin(result.CollectiveId)) return result.info;
+            if (viewer && viewer.id === result.UserId) return result.info;
+            return result.public;
+          });
+        }),
```

Comment:
That should not be modified. There are individual checks on the private properties that make this code not needed. See https://github.com/opencollective/opencollective-api/blob/master/server/graphql/v1/types.js#L662 etc ...